### PR TITLE
Write processing and archive times to DB

### PR DESCRIPTION
**Pull request by yjpa7145**
This addresses CUMULUS-1319: "Granule processing time and archive time are not saved to database/ES"

The `sync_granule_end_time`, `sync_granule_duration`, `post_to_cmr_duration`, and `post_to_cmr_start_time` properties were being written to a a cumulus message's `meta` property _after_ having gone through the Cumulus Message Adapter.  This worked for cases where the CMA returned a full message, but broke when the CMA returned a remote message.  (Those meta values being set in the remote message would be overwritten when the actual message was fetched from S3.)

In addition, the per-granule stats that were being stored for the sync-granule times were actually the times for syncing all of the granules in a task.

This PR stores the timing stats in the granules, instead of in meta, so that they are not lost and are more accurate per-granule.  It also adds a number of tests.

### CODE REVIEW

File: example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js [hunk 1]
```diff
@@ -345,15 +364,28 @@ describe('The S3 Ingest Granules workflow', () => {
       await accessTokensModel.delete({ accessToken });
     });
 
-    it('has expected payload', () => {
-      expect(granule.published).toBe(true);
+    xit('has expected payload', () => {
```
Comment (markdboyd) [inline example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js]: I don't think you want this disabled

Comment (yjpa7145) [reply example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js]: Fixed. I swear, you catch either an `.only` or `xit` in every PR I issue.

File: packages/api/models/granules.js [hunk 2]
```diff
@@ -283,72 +283,72 @@ class Granule extends Manager {
...
+            error: parseException(cumulusMessage.exception),
+            createdAt: get(cumulusMessage, 'cumulus_meta.workflow_start_time'),
+            timestamp: Date.now(),
+            productVolume: getGranuleProductVolume(granuleFiles),
+            timeToPreprocess: get(granule, 'sync_granule_duration', 0) / 1000,
+            timeToArchive: get(granule, 'post_to_cmr_duration', 0) / 1000,
+            processingStartDateTime: granule.sync_granule_end_time
+              ? (new Date(granule.sync_granule_end_time)).toISOString()
+              : undefined,
+            processingEndDateTime: granule.post_to_cmr_start_time
```
Comment (jennyhliu) [inline packages/api/models/granules.js]: If the granule is not published to cmr (no cmr step), I think we still need processingStartDateTime/processingEndDateTime.

**Comment by jennyhliu** [comment]
If you run the example/spec/parallel/emsReport/emsIngestReportSpec.js, and verify that the generated `Ing` report has processingStartDateTime and processingEndDateTime populated.

File: packages/api/models/granules.js (hunk 2 above)
Comment (jennyhliu) [reply packages/api/models/granules.js]: If these properties `sync_granule_end_time, sync_granule_duration, post_to_cmr_duration, and post_to_cmr_start_time` are used for populating `timeToPreprocess, timeToArchive, processingStartDateTime, processingEndDateTime`, should they be removed from granule when they are not needed anymore?

Comment (yjpa7145) [reply packages/api/models/granules.js]: > should they be removed from granule when they are not needed anymore?

This code is not run as part of a workflow, it is triggered by an execution being written to SNS.  The return value is not used anywhere, so there isn't really a reason to modify the input granule.

**Comment by yjpa7145** [comment]
@jennyhliu Where is emsIngestReportSpec.js?  I don't see it on master.

**Comment by jennyhliu** [comment]
> @jennyhliu Where is emsIngestReportSpec.js? I don't see it on master.
Sorry, renamed:
https://github.com/nasa/cumulus/blob/master/example/spec/parallel/emsReport/emsIngestDistributionReportSpec.js

File: packages/api/models/granules.js (hunk 2 above)
Comment (yjpa7145) [reply packages/api/models/granules.js]: > If the granule is not published to cmr (no cmr step), I think the granule should still have processingStartDateTime/processingEndDateTime. These two fields are required by EMS Ing report.

I talked to @bilts about it, and he felt that was out of scope for this ticket.

**Comment by yjpa7145** [comment]
> > @jennyhliu Where is emsIngestReportSpec.js? I don't see it on master.
> > Sorry, renamed:
> > https://github.com/nasa/cumulus/blob/master/example/spec/parallel/emsReport/emsIngestDistributionReportSpec.js

I just ran this test locally and verified that it passed.

**Comment by jennyhliu** [comment]
> > > @jennyhliu Where is emsIngestReportSpec.js? I don't see it on master.
> > > Sorry, renamed:
> > > https://github.com/nasa/cumulus/blob/master/example/spec/parallel/emsReport/emsIngestDistributionReportSpec.js
> 
> I just ran this test locally and verified that it passed.

I ran the test and only the published granule `MOD14A1.A8495082.6XenyA.006.4926687413882` has processingStartDateTime and processingEndDateTime.

MOD14A1.A4042316.GBgfI5.006.8218975078560|&|MOD14A1|&|251784|&|Successful|&|s3_provider_test-jl-test-integration-emsIngestReport-1561125656250|&|2019-06-21 02:02PM|&||&|0|&|0.432|&|21.631
MOD14A1.A8495082.6XenyA.006.4926687413882|&|MOD14A1|&|251784|&|Successful|&|s3_provider_test-jl-test-integration-emsIngestReport-1561125656250|&|2019-06-21 02:02PM|&|2019-06-21 02:02PM|&|9.033|&|0.259|&|38.36

Wrote ticket to address the issue https://bugs.earthdata.nasa.gov/browse/CUMULUS-1362

**Pull request merged**